### PR TITLE
fix for checking group before creating diary, chat

### PR DESCRIPTION
**Pull request by rabsef-bicrym**
#1019 fix

### CODE REVIEW

**Comment by jamesacklin** [comment]
Thank you, @rabsef-bicrym! @liam-fitzgerald will be your friendly reviewer for this.

**Comment by rabsef-bicrym** [comment]
@jamesacklin @liam-fitzgerald thanks - I'm going to push an update that checks if ur in `bloc` too - give us just a few.

image

**Comment by rabsef-bicrym** [comment]
Actually - I'm having an issue modifying chat now - I'm getting a `versioned-state`/`current-state` mismatch - is this expected? is my version bad? i'm working off master.

I will push the fixed diary version now.

**Comment by arthyn** [comment]
@rabsef-bicrym I think we had a bad migration that may have left you in an invalid state. you might need to nuke to fix.

**Review by arthyn — APPROVED** [review]
This LGTM, thank you! Someone mentioned that @liam-fitzgerald wanted to take a look so I'll wait to merge until he's had a chance.

File: desk/app/chat.hoon [hunk 1]
```diff
@@ -256,13 +256,25 @@
   ++  create
     |=  req=create:c
     ^+  cor
+    |^
+    ?.  can-nest
+      %.  cor
+      (slog 'Create failed: group not found' ~)
```
Comment (liam-fitzgerald) [inline desk/app/chat.hoon:262]: naughty! please crash here

**Review by liam-fitzgerald — CHANGES_REQUESTED** [review]
missing relevant changes for heap?
tyfys nice to have outside PRs

File: desk/app/chat.hoon (hunk 1 above)
Comment (rabsef-bicrym) [reply desk/app/chat.hoon]:
```
~_  leaf+"Create failed: check group permissions"
?>  can-nest
```

Now in both diary and chat. Did not know what heap was until like 2 days ago lol. Will resolve.

File: desk/app/chat.hoon [hunk 2]
```diff
@@ -255,14 +255,28 @@
...
+      =/  =perm:c  [writers.req group.req]
+      =.  perm.chat  perm
+      =.  net.chat  [%pub ~]
+      =.  chats  (~(put by chats) flag chat)
+      ca-abet:(ca-init:(ca-abed:ca-core flag) req)
+    ++  can-nest
+      ^-  ?
+      =/  gop  (~(got by groups) group.req)
+      %-  ~(any in bloc.gop)
+      ~(has in sects:(~(got by fleet.gop) our.bowl))    ::  XX: is this right?
```
Comment (liam-fitzgerald) [inline desk/app/chat.hoon:272]: pls remove, this is correct

**Review by liam-fitzgerald — APPROVED** [review]
LGTM! pls remove the `is this right` comments

**Comment by jamesacklin** [comment]
@arthyn @liam-fitzgerald @ryjm Does this affect state? Are we good to merge this?

**Comment by liam-fitzgerald** [comment]
Merge at will

**Comment by rabsef-bicrym** [comment]
I owe you some comment removals. Will have those pushed in 30 min.

**Pull request merged**